lib/vrf.c

```diff
@@ -424,32 +446,86 @@ int vrf_socket(int domain, int type, int protocol, vrf_id_t vrf_id)
 	return ret;
 }
 
-/* vrf CLI commands */
-DEFUN_NOSH (vrf,
-       vrf_cmd,
-       "vrf NAME",
-       "Select a VRF to configure\n"
-       "VRF's name\n")
+extern int vrf_is_backend_ns(void)
+{
+	return vrf_backend_netns;
+}
+
+void vrf_configure_backend(int ns_on)
+{
+	vrf_backend_netns = ns_on;
+}
+
+int vrf_handler_create(struct vty *vty, const char *vrfname, struct vrf **vrf)
 {
-	int idx_name = 1;
-	const char *vrfname = argv[idx_name]->arg;
 	struct vrf *vrfp;
 
 	if (strlen(vrfname) > VRF_NAMSIZ) {
-		vty_out(vty,
-			"%% VRF name %s is invalid: length exceeds "
-			"%d characters\n",
-			vrfname, VRF_NAMSIZ);
+		if (vty)
+			vty_out(vty,
+			    "%% VRF name %s is invalid: length exceeds "
+			    "%d characters\n", vrfname, VRF_NAMSIZ);
+		else
+			zlog_warn("%% VRF name %s is invalid: length exceeds "
+				  "%d characters\n", vrfname, VRF_NAMSIZ);
 		return CMD_WARNING_CONFIG_FAILED;
 	}
 
 	vrfp = vrf_get(VRF_UNKNOWN, vrfname);
 
-	VTY_PUSH_CONTEXT(VRF_NODE, vrfp);
+	if (vty)
+		VTY_PUSH_CONTEXT(VRF_NODE, vrfp);
 
+	if (vrf)
+		*vrf = vrfp;
 	return CMD_SUCCESS;
 }
 
+int vrf_is_mapped_on_netns(struct vrf *vrf)
+{
+	if (!vrf || vrf->data.ns_name[0] == '\0')
+		return 0;
+	if (vrf->vrf_id == NS_DEFAULT)
+		return 0;
+	return 1;
+}
+
+int vrf_switch_to_netns(struct vrf *vrf)
```

Comment:
This function is never used

Reply:
- I made a commit to bring first the APIS.
then BGP commit will come after, and that other commit will use it.
I hope there is no problem in putting useless function at T0, but useful at T1 ?

- vrf_switch_to_netns() is the wrapper for ns_switch_to_netns().